### PR TITLE
tests for reservationTile lwc component

### PR DESCRIPTION
**Pull request by msrivastav13**
### What does this PR do?

tests for reservationTile lwc component

### What issues does this PR fix or reference?

#<Insert GitHub Issue>

## The PR fulfills these requirements:

[ ] Tests for the proposed changes have been added/updated.
[ ] Code linting and formatting was performed.

### Functionality Before

<insert gif and/or summary>

### Functionality After

<insert gif and/or summary>

### CODE REVIEW

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+describe('c-reservation-tile', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+
+    it('render reservationTile for Lead', () => {
```
Comment (muenzpraeger) [inline es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: Small typo `renders`, not `render`.

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        const element = createElement('c-reservation-tile', {
+            is: ReservationTile
+        });
+        element.reservation = RESERVATION_LEAD;
+        document.body.appendChild(element);
+        return Promise.resolve().then(() => {
```
Comment (muenzpraeger) [inline es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js:23]: Would be great to check if you can add here and in other places a few new lines to increase readability.

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            expect(datetimeElements[0].value).toBe(
+                RESERVATION_LEAD.record.Start_Date__c
+            );
+            expect(datetimeElements[1].value).toBe(
+                RESERVATION_LEAD.record.End_Date__c
+            );
+        });
+    });
+
+    it('render reservationTile for Contact', () => {
```
Comment (muenzpraeger) [inline es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: `renders`, same as above

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            expect(datetimeElements[0].value).toBe(
+                RESERVATION_CONTACT.record.Start_Date__c
+            );
+            expect(datetimeElements[1].value).toBe(
+                RESERVATION_CONTACT.record.End_Date__c
+            );
+        });
+    });
+
+    it('click on reservation tile', () => {
```
Comment (muenzpraeger) [inline es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: `fires a reservationselect` CustomEvent on reservationTile click`

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        const element = createElement('c-reservation-tile', {
+            is: ReservationTile
+        });
+        element.reservation = RESERVATION_CONTACT;
+        document.body.appendChild(element);
+        // listen to flow navigate event
```
Comment (muenzpraeger) [inline es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: Flow? Maybe copy-paste error.

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+        const handler = jest.fn();
+        element.addEventListener('reservationselect', handler);
+        return Promise.resolve()
```
Comment (muenzpraeger) [inline es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: No need to put stuff into a Promise here. You would need the Promise only for checking thing after a UI update. As everything is rendered in this component directly when you append the element to the DOM (as in: you don't have any wire or value setting in your component) its not needed.

**Review by muenzpraeger — DISMISSED** [review]

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js (hunk 1 above)
Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: done!

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js (hunk 3 above)
Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: done!

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js (hunk 5 above)
Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: done!

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js (hunk 4 above)
Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: done!

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js (hunk 6 above)
Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: fixed!

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+        const element = createElement('c-reservation-tile', {
+            is: ReservationTile
+        });
+        element.reservation = RESERVATION_LEAD;
+        document.body.appendChild(element);
+
+        return Promise.resolve().then(() => {
+            // check for div class
+            const divElement = element.shadowRoot.querySelector('div.pointer');
+            expect(divElement.className).toBe('pointer');
```
Comment (pozil) [inline es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: This isn't a good test given the fact that you selected the div with a `.pointer` selector.
Update the selector to just `div`.

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+
+            // check for paragraph text nodes
+            const paragraphElements = element.shadowRoot.querySelectorAll('p');
+            expect(paragraphElements[0].textContent).toBe(
+                'Customer: ' + RESERVATION_LEAD.record.Lead__r.Name
```
Comment (pozil) [inline es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: Use string templates everywhere:
```js
`Customer: ${RESERVATION_LEAD.record.Lead__r.Name}`
```

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+    it('renders reservationTile for Contact', () => {
+        const element = createElement('c-reservation-tile', {
+            is: ReservationTile
+        });
+        element.reservation = RESERVATION_CONTACT;
+        document.body.appendChild(element);
+
+        return Promise.resolve().then(() => {
+            // check for div class
+            const divElement = element.shadowRoot.querySelector('div.pointer');
```
Comment (pozil) [inline es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: See comments from previous test.

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+            expect(datetimeElements[0].value).toBe(
+                RESERVATION_CONTACT.record.Start_Date__c
+            );
+            expect(datetimeElements[1].value).toBe(
+                RESERVATION_CONTACT.record.End_Date__c
+            );
+        });
+    });
+
+    it('fires a reservationselect', () => {
```
Comment (pozil) [inline es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: You need a duplicate of this test for a lead tile + be more descriptive in your test name "fires a reservationselect when a contact tile is clicked"

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+        element.reservation = RESERVATION_CONTACT;
+        document.body.appendChild(element);
+
+        // listen to reservationselect event
+        const handler = jest.fn();
+        element.addEventListener('reservationselect', handler);
+        // check for div class
+        const divElement = element.shadowRoot.querySelector('div.pointer');
+        expect(divElement.className).toBe('pointer');
+        divElement.click();
```
Comment (pozil) [inline es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: No need to re-test the class name here. Simply replace the 3 lines with just:
```js
element.shadowRoot.querySelector('div').click();
```

**Review by pozil — CHANGES_REQUESTED** [review]
See comments in code + you're missing a test for muted reservations CSS class.

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js (hunk 7 above)
Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: done!

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js (hunk 8 above)
Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: ah, string literals! How can I forget them! done!

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js (hunk 9 above)
Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: done!

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js (hunk 10 above)
Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: done!

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js (hunk 11 above)
Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: done!

**Comment by msrivastav13** [comment]
@pozil for checking css I have created a separate test case at the bottom! Let me know if that’s not ok and you want me to mix it with first one!

File: es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js (hunk 8 above)
Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js]: done!

**Review by pozil — APPROVED** [review]
All good

**Pull request merged**